JustSaying.AwsTools/SqsNotificationListener.cs

```diff
@@ -269,14 +260,53 @@ public void ProcessMessageAction(Amazon.SQS.Model.Message message)
                     message,
                     ex.StackTrace);
                 Log.Error(ex, msg);
+
                 if (typedMessage != null)
                 {
                     _messagingMonitor.HandleException(typedMessage.GetType().Name);
                 }
+
                 _onError(ex, message);
-                
             }
         }
+
+        private bool CallMessageHandlers(Message message)
+        {
+            List<HandlerFunc> handlerFuncs;
+            var foundHandlers = _handlers.TryGetValue(message.GetType(), out handlerFuncs);
+
+            if (!foundHandlers)
+            {
+                return true;
+            }
+
+            var allHandlersSucceeded = true;
+            foreach (var handlerFunc in handlerFuncs)
+            {
+                var watch = new System.Diagnostics.Stopwatch();
+                watch.Start();
+
+                var thisHandlerSucceeded = handlerFunc(message);
+                allHandlersSucceeded = allHandlersSucceeded && thisHandlerSucceeded;

```

Comment:
Agreed that " deleting messages that have no handlers" is questionable, but should be in a separate PR.